lib/p2p/Pool.ts

```diff
@@ -86,19 +104,55 @@ class Pool extends EventEmitter {
   }
 
   /**
-   * Attempt to add an outbound peer by connecting to a given SocketAddress.
-   * Throws an error if a connection already exists for the provided address.
-   * If the connection is successful, send a [[GetHostsPacket]] to discover new XUD hosts to connect to.
+   * Iterate over a collection of nodes and attempt to connect to them.
+   * If the node is banned, already connected, or has no listening addresses, then do nothing.
+   * @param nodes a collection of nodes with a `forEach` iterator to attempt to connect to
+   * @param ignoreKnown whether to ignore nodes we are already aware of, defaults to false
+   * @returns a promise that will resolve when all outbound connections resolve
+   */
+  private connectNodes = (nodes: NodeConnectionIterator, ignoreKnown = false) => {
+    const connectionPromises: Promise<void>[] = [];
+    nodes.forEach((node) => {
+      // check that this node is not ourselves, that it has listening addresses,
+      // and that either we haven't heard of it, or we're not ignoring known nodes and it's not banned
+      if (node.nodePubKey !== this.handshakeData.nodePubKey && node.addresses.length > 0 &&
+        (!this.nodes.has(node.nodePubKey) || (!ignoreKnown && !this.nodes.isBanned(node.nodePubKey)))) {
+        connectionPromises.push(this.connectNode(node));
+      }
+    });
+    return Promise.all(connectionPromises);
+  }
+
+  /**
+   * Attempt to create an outbound connection to a node using its known listening addresses.
+   */
+  private connectNode = async ({ addresses, nodePubKey }: NodeConnectionInfo) => {
+    for (let n = 0; n < addresses.length; n += 1) {
```

Comment:
`forEach` instead? parallel to `connectNodes`

Reply:
The reason I used a regular for loop is to be able to use the `break` command after the first successful connection, which I don't believe works with `.forEach()`.